### PR TITLE
refactor: change defined(MAS_BUILD) to IS_MAS_BUILD()

### PR DESCRIPTION
**Pull request by MarshallOfSound**
This is missing-definition safe and thus allows us to move the definition of this macro away from "all compilation targets" to "just the compilation targets that depend on this macro".

In turn this makes the rebuild time changing from mas <-> darwin only 80 seconds on my machine, instead of the 12-15 minutes it used to take.  This will also allow us in the future to build both MAS and darwin on the same CI machine.  Costing us ~2 minutes on one machine but saving us anywhere from 30 minutes to an hour of CI time on other parts of the matrix.

I think before this lands I want to clean up the new patch file a bit (potentially merge those changes into the other MAS patch files) but I'll consider that tomorrow.

Notes: no-notes

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -977,6 +978,11 @@ if (is_mac) {
         deps += [ "//sandbox/mac:seatbelt" ]
       }
       defines = [ "HELPER_EXECUTABLE" ]
+      if (is_mas_build) {
+        defines += [ "IS_MAS_BUILD()=1" ]
+      } else {
+        defines += [ "IS_MAS_BUILD()=0" ]
+      }
```
Comment (nornagon) [inline BUILD.gn:985]: how come these are duplicated from the config?

File: patches/chromium/build_only_use_the_mas_build_config_in_the_required_components.patch [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+   if (is_android || use_ozone) {
+diff --git a/content/browser/BUILD.gn b/content/browser/BUILD.gn
+index b33864b2d57fb26637bd96edbd8fb917881f5f4c..749949d1b3618a16fce3a47dd2776af3555f64d2 100644
+--- a/content/browser/BUILD.gn
++++ b/content/browser/BUILD.gn
+@@ -52,6 +52,7 @@ source_set("browser") {
+     "//tools/v8_context_snapshot:use_v8_context_snapshot",
+     "//v8:external_startup_data",
+   ]
++  configs += ["//electron/build/config:mas_build"]
```
Comment (nornagon) [inline patches/chromium/build_only_use_the_mas_build_config_in_the_required_components.patch:78]: //content/browser is a big component, any chance we could narrow this?

**Review by nornagon — APPROVED** [review]
lgtm, gosh we patch a lot for MAS

File: patches/chromium/build_only_use_the_mas_build_config_in_the_required_components.patch (hunk 2 above)
Comment (MarshallOfSound) [reply patches/chromium/build_only_use_the_mas_build_config_in_the_required_components.patch:78]: Unfortunately not, I tried to find a nearer `BUILD.gn` and couldn't

File: BUILD.gn (hunk 1 above)
Comment (nornagon) [reply BUILD.gn:985]: i don't think that's true? https://source.chromium.org/chromium/chromium/src/+/main:build/config/mac/rules.gni;l=479-498;drc=88d2dd9aa6fe5071c862699b53aab45bd8a5ad93

either `configs` or `extra_configs` should work.

File: patches/chromium/build_only_use_the_mas_build_config_in_the_required_components.patch (hunk 2 above)
Comment (nornagon) [reply patches/chromium/build_only_use_the_mas_build_config_in_the_required_components.patch:78]: rip

File: BUILD.gn (hunk 1 above)
Comment (MarshallOfSound) [reply BUILD.gn:985]: Oh hm `extra_configs`, I didn't try that one lol

**Pull request merged**